sequencer/src/block2.rs

```diff
@@ -555,6 +627,83 @@ mod tx_table_entry {
     }
 }
 
+type NsTable = <BlockPayload as hotshot::traits::BlockPayload>::Metadata;
+
+/// TODO do we really need `PartialOrd`, `Ord` here?
+/// Could the `Ord` bound be removed from `QueryablePayload::TransactionIndex`?`
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+pub struct TxIndex {
+    ns_idx: usize,
+    tx_idx: usize,
+}
+
+pub struct TxIterator<'a> {
+    ns_idx: usize, // TODO should be able to eliminate this field
+    ns_iter: Range<usize>,
+    tx_iter: Range<usize>,
+    block_payload: &'a BlockPayload,
+
+    // TODO should be `&'a NsTable`, but that requires upstream change to `QuearyablePayload::iter`:
+    // fn iter<'a>(&'a self, meta: &'a Self::Metadata)
+    //        ++++  ++              ++
+    ns_table: NsTable,
```

Comment:
@jbearer looks like we need another tweak to `QueryablePayload` 😕

Reply:
Ah, that looks like a good change. I can do it, should be quick 🤞